contributors/design-proposals/add-metadata-indicating-to-replace-union-in-patch.md

```diff
@@ -0,0 +1,280 @@
+Add Tag Indicating To Replace *Union* Fields in Patch (a.k.a Support Replacing Maps in Strategic Merge Patch)
+=============
+
+This approach could be used to fix the issues with `kubectl apply` simply by having apply always replace union fields instead of patching them.  E.g. instead of setting `spec.strategy.type` without touching the rest of `spec.strategy`, patches should always replace the entire `spec.strategy`.
+
+## Proposed Changes
+
+### APIs
+
+Add a go struct tag/openAPI value telling clients to replace the entire struct when doing a *PATCH*. For the *inline* fields, we have 2 options:
+
+1) Move the tag to the parent struct's. E.g. add a tag to `Volume` instead of `VolumeSource`. This will limit the flexibility to add more fields in the future.
+
+2) Add some logic when lookup the metadata in go struct to find the tags for inline fields.
```

Comment:
I believe that is ok for the current use cases, we can just put it in the parent right?